File: idscp2/src/main/java/de/fhg/aisec/ids/idscp2/idscp_core/configuration/AttestationConfig.java
package de.fhg.aisec.ids.idscp2.idscp_core.configuration;

import java.util.Arrays;

/**
 * Attestation configuration class, containing attestation suite for supported / expected
 * attestation types
 *
 * @author Leon Beckmann (leon.beckmann@aisec.fraunhofer.de)
 */
public class AttestationConfig {
    public final static String[] DEFAULT_RAT_MECHANISMS = new String[] {"Dummy", "TPM2d"};
    private final String[] ratMechanisms;

    public AttestationConfig() {
        this.ratMechanisms = DEFAULT_RAT_MECHANISMS;
    }

    public String[] getRatMechanisms() {
        return ratMechanisms;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        AttestationConfig that = (AttestationConfig) o;
        return Arrays.equals(ratMechanisms, that.ratMechanisms);
    }

    @Override
    public int hashCode() {
        return Arrays.hashCode(ratMechanisms);
    }
}

File: idscp2/src/main/kotlin/de/fhg/aisec/ids/idscp2/idscp_core/configuration/AttestationConfig.kt
package de.fhg.aisec.ids.idscp2.idscp_core.configuration

/**
 * Attestation configuration class, containing attestation suite for supported / expected
 * attestation types
 *
 * @author Leon Beckmann (leon.beckmann@aisec.fraunhofer.de)
 */
class AttestationConfig {
    val ratMechanisms: Array<String>
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as AttestationConfig
        return ratMechanisms.contentEquals(that.ratMechanisms)
    }

    override fun hashCode(): Int {
        return ratMechanisms.contentHashCode()
    }

    companion object {
        val DEFAULT_RAT_MECHANISMS = arrayOf("Dummy", "TPM2d")
    }

    init {
        ratMechanisms = DEFAULT_RAT_MECHANISMS
    }
}